pysuccinct/pyquery_file.cpp

```diff
@@ -0,0 +1,98 @@
+#include <cstdio>
+#include <iostream>
+#include <unistd.h>
+#include <ctime>
+#include <sstream>
+#include <sys/time.h>
+
+#include "succinct_file.h"
+
+#include <boost/python.hpp>
+using namespace boost::python;
+
+/**
+ * Prints usage.
+ */
+void print_usage(char *exec) {
+  fprintf(stderr, "Usage: %s [-m mode] [file]\n", exec);
+}
+
+void print_valid_cmds() {
+  std::cerr
+      << "Command must be one of: search [query], count [query], extract [offset] [length]\n";
+}
+
+typedef unsigned long long int timestamp_t;
+
+static timestamp_t get_timestamp() {
+  struct timeval now{};
+  gettimeofday(&now, nullptr);
+
+  return (now.tv_usec + (time_t) now.tv_sec * 1000000);
+}
+
+struct QueryFile {
+    QueryFile(std::string filename, uint32_t mode){
+        this->s_file = nullptr;
+        if (mode == 0) {
+            // If mode is set to 0, compress the input file.
+            // Use default parameters.
+            std::cout << "Constructing Succinct data structures...\n";
+            this->s_file = new SuccinctFile(filename);
+            std::cout << "Serializing Succinct data structures...\n";
+            this->s_file->Serialize(filename + ".succinct");
+        } else {
+            // If mode is set to 1, read the serialized data structures from disk.
+            // The serialized data structures must exist at <filename>.succinct.
+            std::cout << "De-serializing Succinct data structures...\n";
+            this->s_file = new SuccinctFile(filename, SuccinctMode::LOAD_IN_MEMORY);
+        }
+        std::cout << "Done. Starting Succinct Shell...\n";
+        print_valid_cmds();
+    }
+
+    //QueryFile members
+    SuccinctFile *s_file;
+
+    //Wrapped search command
+    void search(std::string arg){
```

Comment:
Pass objects (e.g., strings) by const reference
```suggestion
    void search(const std::string& arg) {
```